oml/lightning/callbacks/metric.py

```diff
@@ -72,6 +89,30 @@ def on_validation_batch_end(
             if is_last_expected_batch:
                 self.calc_and_log_metrics(pl_module)
 
+    def _log_images(self, pl_module: pl.LightningDataModule) -> None:
+        if not isinstance(self.metric, EmbeddingMetrics):
+            return
+
+        for metric_name in flatten_dict(self.metric.metrics):  # type: ignore
```

Comment:
We have `IBasicMetric` which guide us support following structure

```
{
            "self.overall_categories_key": {"metric1": v1, "metric2": v2},
            "category1": {"metric1": v1, "metric2": v2},
            "category2": {"metric1": v1, "metric2": v2}
        }
```

Probably we need to iterate in this structure instead of `flatten_dict`?